terraform/core/47-mwaa.tf

```diff
@@ -0,0 +1,187 @@
+          "logs:GetLogRecord",
+          "logs:GetLogGroupFields",
+          "logs:GetQueryResults",
+          "logs:DescribeLogGroups"
+        ],
+        Resource = "arn:aws:logs:*"
+      },
+      {
+        Effect   = "Allow",
+        Action   = "cloudwatch:PutMetricData",
+        Resource = "*"
+      },
+      {
+        Effect = "Allow",
+        Action = [
+          "sqs:ChangeMessageVisibility",
+          "sqs:DeleteMessage",
+          "sqs:GetQueueAttributes",
+          "sqs:GetQueueUrl",
+          "sqs:ReceiveMessage",
+          "sqs:SendMessage"
+        ],
+        Resource = "arn:aws:sqs:eu-west-2:*:airflow-celery-*"
+      },
+      {
+        Effect = "Allow",
+        Action = [
+          "kms:Decrypt",
+          "kms:DescribeKey",
+          "kms:GenerateDataKey*",
+          "kms:Encrypt"
+        ],
+        Resource = aws_kms_key.mwaa_key.arn
+      }
+    ]
+  })
+}
+
+
+# Security group for MWAA - self-referencing and allowing all traffic out
+# This is recommended in the doc, Matt recommended at current stage.
+# https://docs.aws.amazon.com/mwaa/latest/userguide/vpc-security.html
+resource "aws_security_group" "mwaa_sg" {
+  name        = "mwaa_sg"
+  description = "Security group for MWAA"
+  vpc_id      = data.aws_vpc.network.id
+  tags        = module.tags.values
+}
+
+resource "aws_security_group_rule" "mwaa_ingress" {
+  type                     = "ingress"
+  from_port                = 0
+  to_port                  = 0
+  protocol                 = "-1"
+  security_group_id        = aws_security_group.mwaa_sg.id
+  source_security_group_id = aws_security_group.mwaa_sg.id
+  description              = "Allow all inbound traffic from specific security group"
+}
+
+resource "aws_security_group_rule" "mwaa_egress" {
+  type              = "egress"
+  from_port         = 0
+  to_port           = 0
+  protocol          = "-1"
+  security_group_id = aws_security_group.mwaa_sg.id
+  cidr_blocks       = ["0.0.0.0/0"]
+  description       = "Allow all outbound traffic"
+}
+
+# Create a placeholder file in the S3 bucket, S3 does not support the creation of empty folders
+# To ensure the folders are created otherwise the MWAA environment will fail to create
+resource "aws_s3_bucket_object" "plugins_placeholder" {
+  # bucket  = module.mwaa_bucket.bucket_id
+  bucket  = aws_s3_bucket.mwaa_bucket.bucket
+  key     = "plugins/.placeholder"
+  content = ""
+}
+
+resource "aws_s3_bucket_object" "dags_placeholder" {
+  bucket  = aws_s3_bucket.mwaa_bucket.bucket
+  key     = "dags/.placeholder"
+  content = ""
+}
+
+resource "aws_s3_bucket_object" "requirements_placeholder" {
+  bucket  = aws_s3_bucket.mwaa_bucket.bucket
+  key     = "requirements/.placeholder"
+  content = ""
+}
+
+resource "aws_mwaa_environment" "mwaa" {
+  # count              = local.is_live_environment && !local.is_production_environment ? 1 : 0
+  name               = "dataplatform-${local.short_identifier_prefix}mwaa-environment"
```

Comment:
Hey Tim, same as the bucket name. Is it better to use `${local.identifier_prefix}-mwaa-environment` here? Thanks @timburke-hackit

Reply:
yeah, probably - it's not a big deal but maybe makes it easier to have all the mwaa resources named consistently?